datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemBooleanTypePickerViewHolderFactoryInstrumentedTest.kt

```diff
@@ -322,16 +322,46 @@ class QuestionnaireItemBooleanTypePickerViewHolderFactoryInstrumentedTest {
 
   @Test
   @UiThreadTest
-  fun displayValidationResult_error_shouldShowErrorMessage() {
+  fun displayValidationResult_questionnaireResponseAnswerWithValue_shouldShowNoErrorMessage() {
     viewHolder.bind(
       QuestionnaireItemViewItem(
-        Questionnaire.QuestionnaireItemComponent().apply { required = true },
+        Questionnaire.QuestionnaireItemComponent().apply {
+          required = true
+          addAnswerOption(
+            Questionnaire.QuestionnaireItemAnswerOptionComponent().apply { text = "Question?" }
+          )
+        },
+        QuestionnaireResponse.QuestionnaireResponseItemComponent().apply {
+          addAnswer(
+            QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent().apply {
+              value = BooleanType(false)
+            }
+          )
+        }
+      ) {}
+    )
+
+    assertThat(viewHolder.itemView.findViewById<TextView>(R.id.error_text_view).error)
+      .isEqualTo(null)
+  }
+
+  @Test
+  @UiThreadTest
+  fun displayValidationResult_noQuestionnaireResponseAnswer_shouldShowNoErrorMessage() {
+    viewHolder.bind(
+      QuestionnaireItemViewItem(
+        Questionnaire.QuestionnaireItemComponent().apply {
+          required = true
+          addAnswerOption(
+            Questionnaire.QuestionnaireItemAnswerOptionComponent().apply { text = "Question?" }
+          )
+        },
         QuestionnaireResponse.QuestionnaireResponseItemComponent()
       ) {}
     )
 
-    assertThat(viewHolder.itemView.findViewById<TextView>(R.id.error_text_view).text)
-      .isEqualTo("Missing answer for required field.")
+    assertThat(viewHolder.itemView.findViewById<TextView>(R.id.error_text_view).error)
```

Comment:
i think you need to assert on `.text` rather than `.error`

```suggestion
    assertThat(viewHolder.itemView.findViewById<TextView>(R.id.error_text_view).text)
```